service/src/main/java/com/myhome/security/filters/CommunityAuthorizationFilter.java

```diff
@@ -0,0 +1,66 @@
+package com.myhome.security.filters;
+
+import com.myhome.domain.User;
+import com.myhome.services.CommunityService;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CommunityAuthorizationFilter extends BasicAuthenticationFilter {
+    private final CommunityService communityService;
+    private final String uuidPattern = "[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}";
+    private final Pattern addAmenityRequestPattern = Pattern.compile("/communities/" + uuidPattern + "/amenities");
+
+
+    public CommunityAuthorizationFilter(AuthenticationManager authenticationManager,
+                                        CommunityService communityService) {
+        super(authenticationManager);
+        this.communityService = communityService;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response,
+                                    FilterChain chain) throws IOException, ServletException {
+
+        Matcher urlMatcher = addAmenityRequestPattern.matcher(request.getRequestURI());
+
+        if (urlMatcher.find() && !isUserCommunityAdmin(request)) {
+            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+            return;
+        }
+
+        super.doFilterInternal(request, response, chain);
+    }
+
+    private boolean isUserCommunityAdmin(HttpServletRequest request) {
+        String userId = (String) SecurityContextHolder
+                .getContext().getAuthentication().getPrincipal();
+        String communityId = request
+                .getRequestURI().split("/")[2];
+        Optional<List<User>> optional = communityService
+                .findCommunityAdminsById(communityId, null);
+
+        if (optional.isPresent()) {
```

Comment:
How about:
```
return communityService.findCommunityAdminsById(communityId, null)
        .flatMap(admins -> admins.stream()
            .map(User::getUserId)
            .filter(userId::equals)
            .findFirst()
        )
        .isPresent();
```

@jmprathab 
For the future, it would be great to store the admin role in jwt token and work on jwt token payload in filters to not call the database every time

Reply:
Store admin role into the token can be little tricky. Admin role is not applying in general, but rather to a specific community. We would have to store every community in token, where the user got admin role. 

In addition imagine the scenario where user will be added into the admin list for particular community and content of token will not be uploaded until the next authentication.

Reply:
Yes, you are right @fx0000 :)